unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldEntityHandler.cs

```diff
@@ -40,6 +41,24 @@ public class BuilderInWorldEntityHandler : MonoBehaviour
 
     float nextTimeToUpdateTransform = 0;
 
+    public void Init()
+    {
+        HUDController.i.buildModeHud.OnEntityDelete += DeleteEntity;
+        HUDController.i.buildModeHud.OnDuplicateSelectedAction += DuplicateSelectedEntitiesInput;
+        HUDController.i.buildModeHud.OnDeleteSelectedAction += DeleteSelectedEntitiesInput;
+        HUDController.i.buildModeHud.OnEntityClick += ChangeEntitySelectionFromList;
+        HUDController.i.buildModeHud.OnEntityLock += ChangeEntityLockStatus;
+        HUDController.i.buildModeHud.OnEntityChangeVisibility += ChangeEntityVisibilityStatus;
+        HUDController.i.buildModeHud.OnEntityRename += ChangeEntityName;
+
+        HUDController.i.buildModeHud.OnSelectedObjectPositionChange += UpdateEntityPosition;
+        HUDController.i.buildModeHud.OnSelectedObjectRotationChange += UpdateEntityRotation;
+        HUDController.i.buildModeHud.OnSelectedObjectScaleChange += UpdateEntityScale;
```

Comment:
I'm worried about coupling the `HUDController` too much. We should try to move this to a data driven approach in the future. I would even prefer an ugly `FindObjectOfType` (assuming `buildModeHud` is a `Monobehaviour`).

Anyway, let's leave it for another PR!

Reply:
It is not a MonoBehaviour, however, I agree that right now the HUDController have a lot of responsibility, I will like to talk about the direction of this controller